### DISCUSSION TITLE
Creating a Whisper pipeline from local checkpoint

### DISCUSSION THREAD

**Discussion by DiwakarBasnet** [openai/whisper#2102] [Q&A] [answered]
While creating a whisper huggingface pipeline from local checkpoint, how to save the trainer? Usually we would save model by,
`trainer.save_model("path/to/local/dir")` but how to save model along with feature extractor and tokenizer for a whisper model, so that a pipeline can be created by just uploading the local checkpoint.

**Comment by Atefeh197** [comment]
Hi 

Did you find any way for that?

**Reply by DiwakarBasnet** [reply, marked as answer]
Hello, I first saved the WhisperProcessor before training by using `processor.save_pretrained("path/to/local/dir")` then after the training is completed I saved the trained model by using `trainer.save_model("path/to/local/dir")`. Then for using it as a pipeline I used `pipeline("automatic-speech-recognition", model="path/to/local/dir")`
